### PR TITLE
Rename some classes from Bing* to Push* to match server denomination

### PR DESCRIPTION
**Pull request by bmarty**

### CODE REVIEW

**Comment by bmarty** [comment]
(no impact on Riot)

File: matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/PushRulesRestClient.java [hunk 1]
```diff
@@ -17,32 +18,28 @@
...
 
     /**
-     * Retrieve the bing rules list.
+     * Retrieve the push rules list.
      *
      * @param callback the asynchronous callback.
      */
-    public void getAllBingRules(final ApiCallback<BingRulesResponse> callback) {
-        mApi.getAllBingRules().enqueue(new DefaultRetrofit2CallbackWrapper<>(callback));
+    public void getAllRules(final ApiCallback<PushRulesResponse> callback) {
```
Comment (giomfo) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/PushRulesRestClient.java]: I would rename it `getAllPushRules`

File: matrix-sdk/src/main/java/org/matrix/androidsdk/rest/api/PushRulesApi.java [hunk 2]
```diff
@@ -26,18 +27,20 @@
...
-public interface BingRulesApi {
+public interface PushRulesApi {
 
-    // TODO Add Javadoc
+    /**
+     * Get all push rules
+     */
     @GET("pushrules/")
-    Call<BingRulesResponse> getAllBingRules();
+    Call<PushRulesResponse> getAllRules();
```
Comment (giomfo) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/rest/api/PushRulesApi.java]: I would rename it getAllPushRules()

File: matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/PushRulesRestClient.java (hunk 1 above)
Comment (bmarty) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/PushRulesRestClient.java]: Other methods are like this (ex: `deleteRule()`) so I think it's more coherent.

File: matrix-sdk/src/main/java/org/matrix/androidsdk/rest/api/PushRulesApi.java (hunk 2 above)
Comment (bmarty) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/rest/api/PushRulesApi.java]: Other methods are like this (ex: `deleteRule()`) so I think it's more coherent.

**Pull request merged**